CleverRaven/Cataclysm-DDA
# palette parameters + roof for saltworks

#### Summary
None



#### Purpose of change
Add explicit roofs to saltworks expansion to remove reliance of roof adding magic.



#### Describe the solution
- Convert the expansion to use palette parameters (as much testing regardless, and probably less work overall, so might as well do it).
- Add explicit roofs plus a chimney.



#### Describe alternatives you've considered
- Remove this vile magic expansion completely.
- Add concrete material.



#### Testing
- Load suitable save, create expansion.
- Build everything out of the first material.
- Inspect results.
- Restart.
- Repeat for all other materials, skipping the still, as it's a common element already covered with the first set.
![Screenshot (388)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/e286409f-f61e-44d9-bad3-088ac647da68)
![Screenshot (389)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/ee3214f2-03a4-4183-b02c-1560def9941d)
![Screenshot (390)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/43463ef4-9840-457d-b2a4-90e770a6499f)
![Screenshot (391)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/e51af9c3-13c7-43a7-814e-3f3d3d4f17eb)
![Screenshot (392)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/ea0cb1dd-9756-4e8d-b7fc-54c659356a6c)
![Screenshot (393)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/715bd491-b7f8-4fdc-91a3-3a87fd8a066b)
![Screenshot (394)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/7f2bf4a8-db82-464f-b587-9a91e2fcd5bd)
![Screenshot (395)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/809d6445-8159-46a4-86fa-f84329db4070)
![Screenshot (396)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/ed69148d-60e1-446e-a175-ed6acb7df57c)
![Screenshot (397)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/76618f55-bf7f-41f4-8cdf-7d58a9731d70)
![Screenshot (398)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/73e6d401-9707-46c1-bd18-b48dd2ec50c8)
![Screenshot (399)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/54c99028-9efb-46c3-8c29-7bb86ca84779)
![Screenshot (400)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/5d3cf2a2-c14d-4b41-8017-defe37d4c13d)
![Screenshot (401)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/b057de2c-b93f-4900-851e-49522fa74a4f)

- Realized I should have used a vent pipe rather than a chimney with the stove. Changed that.
![Screenshot (402)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/35c212fb-6794-421e-a5ce-20901dd2625e)



#### Additional context
